mosspy/download_report.py

```diff
@@ -49,7 +50,7 @@ def process_url(url, urls, base_url, path):
     f.write(soup.encode(soup.original_encoding))
     f.close()
 
-def download_report(url, path, connections = 4, log_level=logging.DEBUG):
+def download_report(url, path, connections = 4, log_level=logging.DEBUG, on_read=lambda url: None):
```

Comment:
Consider making the default follow the very basic example implementation you gave? Then everyone will see some basic form of progress.